### PR TITLE
NAV - Add course page to teacher dashboard with sidebar

### PR DESCRIPTION
**Pull request by lfryemason**
Adds the course page to sidebar

https://github.com/user-attachments/assets/b909c066-02e7-4e13-a1c0-dd29cd11f6c1

## Links
[TEACH-1237](https://codedotorg.atlassian.net/browse/TEACH-1237)

## Follow-up work

* Switch between unit and course page from navbar links
* Add the unit page as well
* Loading state for the course page - currently it is blank while it loads.
* Loading state if loading a different section
* Better handling for when sections are flipped between while course summary is loading

### CODE REVIEW

File: apps/src/sites/studio/pages/teacher_dashboard/show.js [hunk 1]
```diff
@@ -55,6 +59,10 @@ $(document).ready(function () {
     currentUser,
     sectionStandardsProgress,
     locales,
+    viewAs,
```
Comment (lfryemason) [inline apps/src/sites/studio/pages/teacher_dashboard/show.js:62]: Needed to add some reducers that the course page relies on and were not already in teacher_dashboard

File: apps/src/templates/teacherNavigation/TeacherNavigationRouter.tsx [hunk 2]
```diff
@@ -91,197 +96,217 @@ const TeacherNavigationRouter: React.FC<TeacherNavigationRouterProps> = ({
...
-    <Route
-      path={TEACHER_NAVIGATION_SECTIONS_URL}
-      element={
-        <div className={styles.pageAndSidebar}>
-          <TeacherNavigationBar />
-          <Outlet />
-        </div>
-      }
-    >
+  const routes = React.useMemo(
```
Comment (lfryemason) [inline apps/src/templates/teacherNavigation/TeacherNavigationRouter.tsx:99]: memoizing the routes so it doesn't re-render too often.

File: apps/src/templates/teacherNavigation/TeacherNavigationRouter.tsx [hunk 3]
```diff
@@ -91,197 +96,217 @@ const TeacherNavigationRouter: React.FC<TeacherNavigationRouterProps> = ({
...
+            path={TEACHER_NAVIGATION_PATHS.courseOverview}
+            loader={teacherCourseOverviewLoader}
+            element={
+              <ElementOrEmptyPage
+                showNoStudents={false}
+                showNoCurriculumAssigned={
+                  !!selectedSection &&
+                  !selectedSection.courseVersionName &&
+                  !selectedSection.courseOfferingId
+                }
```
Comment (lfryemason) [inline apps/src/templates/teacherNavigation/TeacherNavigationRouter.tsx:257]: This is the only actual change to router

**Review by kobryan0619 — APPROVED** [review]
Looks good!

**Pull request merged**